doc/sphinx/proofs/writing-proofs/inductive.rst

```diff
@@ -96,10 +95,21 @@ Case analysis and induction
 The tactics presented in this section implement induction or case
 analysis on inductive or co-inductive objects (see :ref:`inductive-definitions`).
 
-.. tacn:: destruct @term
-   :name: destruct
+.. tacn:: destruct @induction_clause_list
+
+   .. insertprodn induction_clause_list opt_clause
+
+   .. prodn::
+      induction_clause_list ::= {+, @induction_clause } {? using @one_term {? with @bindings } } {? @opt_clause }
+      induction_clause ::= @destruction_arg {? @as_or_and_ipat } {? eqn : @naming_intropattern } {? @opt_clause }
+      opt_clause ::= in @goal_occurrences
+      | at @occs_nums
+
+   .. todo PR: need to explain how to get @term from @induction_clause_list
+      and @destruction_arg for several tactics,
+      also what it means if multiple @induction_clauses are given
```

Comment:
@JasonGross So I guess you actually use it in practice and would not like to see this form deprecated? Because that's something that I've wanted to do for some time. (I'm talking about `induction x, y` here, not `destruct x, y`.)

BTW do you understand what `induction x, y using foo` does? I was expecting that the induction principle passed with the `using` clause would only be used for the first variable (consistently with the behavior when the `using` clause is not there) but if that was the case, then it wouldn't fail here:

```coq
Goal forall n m, n + m = m + n.
induction n, m using nat_ind. (* Error: Don't know where to find some argument. *)
```

Reply:
> So I guess you actually use it in practice

Yes

>  and would not like to see this form deprecated?

Eh, I won't object strongly, `induction x; destruct y` is not that hard to type.  But I do have a slight preference for keeping it.


> BTW do you understand what `induction x, y using foo` does? I was expecting that the induction principle passed with the `using` clause would only be used for the first variable (consistently with the behavior when the `using` clause is not there) but if that was the case, then it wouldn't fail here:
> 
> ```coq
> Goal forall n m, n + m = m + n.
> induction n, m using nat_ind. (* Error: Don't know where to find some argument. *)
> ```

Heh, I don't.  Even more strange is that you get a different and nonsensically incomplete error message if you run intros first:
```coq
Goal forall n m, n + m = m + n.
  intros.
  destruct n, m using nat_ind. (* Error: Not the right number of induction arguments (expected ). *)
```
Interestingly, I can get the same message with
```coq
Goal forall n m, n + m = m + n.
  intros.
  destruct n using (nat_case _). (* Error: Not the right number of induction arguments (expected ). *)
```

Reply:
Reported this issue as https://github.com/coq/coq/issues/14454.  I suspect it may be related to the issue you see without `intros`.  Perhaps @herbelin knows?

Reply:
> Reported this issue as #14454. I suspect it may be related to the issue you see without `intros`. Perhaps @herbelin knows?

Here is an example of application:
```
Require Import Arith.
Goal forall a b, a +b =0.
intros.
induction a, b using lt_wf_double_rect.
(*
  b1, b2 : nat
  H : forall p q : nat, p < b1 -> p + q = 0
  H0 : forall p : nat, p < b2 -> b1 + p = 0
  ============================
  b1 + b2 = 0
*)
```
When using `using`, as many arguments as the number of arguments of the scheme is expected (plus a special rule to simulate `functional induction` which uses `using` under the hood).

Reply:
Is this also a bug (for `induction a, b.`?)

```
Goal forall a b, a + b = 0.
induction a, b using nat_ind.  (* Don't know where to find some argument. *)


induction a, b.
(*
4 subgoals
______________________________________(1/4)
0 + 0 = 0
______________________________________(2/4)
0 + S b = 0
______________________________________(3/4)
S a + 0 = 0
______________________________________(4/4)
S a + S b = 0
*)

(* FWIW, not equivalent to this: *)
induction a. induction b.
(* 3 subgoals
______________________________________(1/3)
0 + 0 = 0
______________________________________(2/3)
0 + S b = 0
______________________________________(3/3)
forall b : nat, S a + b = 0 *)
```

Reply:
As discussed, `induction x, y` and `induction x, y using ...` really have nothing to do with each other. While the first is a shortcut for `induction x; destruct y`, the second expects an induction principle which talks about predicates of two variables (like in Hugo's example). I'm not a fan of this discrepancy, but since it exists (and is being relied on), let's document it.